server/src/main/java/org/opensearch/search/pipeline/PipelineProcessingContext.java

```diff
@@ -17,6 +20,9 @@
 public class PipelineProcessingContext {
     private final Map<String, Object> attributes = new HashMap<>();
 
+    // Key for processor execution details
+    private static final String PROCESSOR_EXECUTION_DETAILS_KEY = "processorExecutionDetails";
```

Comment:
Something seems off to me about having this private constant in this class.  I see it's used in a getter so nobody needs to know what it is, but it's also a key in an attributes map that could have a name collision.

Are other attributes defined (should this constant be somewhere else, and public)?  If not, is re-using this map for an undocumented purpose a good idea?

Reply:
I noticed a similar usage in the [ScriptRequestProcessor class](https://github.com/opensearch-project/OpenSearch/blob/7ae66d0d63308b009927647ce0a892231f1fad29/modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/ScriptRequestProcessor.java#L149C9-L149C127), where keys are defined as:
```
private static final List<String> SCRIPT_CONFIG_KEYS = List.of("id", "source", "inline", "lang", "params", "options");
```
In our case, I’m considering changing the constant to public and moving it to the ProcessorExecutionDetail class 

Do you think this would be a better approach?